src/routes/user/updateUser.ts

```diff
@@ -0,0 +1,91 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { knexController } from '../../database/database';
+import { hashPassword } from '../../lib/auth';
+import {
+  User,
+  editUserResponse,
+  user,
+  UserParamsPayload,
+  HashObj,
+} from '../../types/user.types';
+
+const editUserSchema = {
+  description: 'Edit data of already existing user.',
+  summary: 'Edit user',
+  tags: ['User', 'Update'],
+  body: {
+    type: 'object',
+    required: [],
+    properties: user.static,
+  },
+  response: {
+    200: editUserResponse,
+    500: { $ref: 'error' },
+    400: { $ref: 'error' },
+    404: { $ref: 'error' },
+  },
+};
+
+const editUserHandler = async (
+  req: FastifyRequest<{ Body: User; Params: UserParamsPayload }>,
+  reply: FastifyReply
+): Promise<void> => {
+  const { email, password, forename, surname, isAdmin, isBlender } = req.body;
+
+  const userId = req.params.userId;
+
+  let hashObj: HashObj = {
+    hash: undefined,
+    salt: undefined,
+  };
```

Comment:
Pahoittelut en sanonu ihan tarkasti mitä meinasin, mutta tää siis voisi olla

```suggestion
let hashObj: HashObj | undefined = undefined;
```

Reply:
Ei käsittääkseni ole tilannetta missä voisi olla hash undefined ja salt ei ole tai toisinpäin